app/stac_api/models.py

```diff
@@ -483,6 +487,31 @@ class Item(models.Model):
 
     class Meta:
         unique_together = (('collection', 'name'),)
+        indexes = [
+            models.Index(fields=['name'], name='item_name_idx'),
+            models.Index(fields=['properties_datetime'], name='item_datetime_idx'),
+            models.Index(fields=['properties_start_datetime'], name='item_start_datetime_idx'),
+            models.Index(fields=['properties_end_datetime'], name='item_end_datetime_idx'),
+            BrinIndex(fields=['created'], name='item_created_idx'),
+            models.Index(fields=['updated'], name='item_updated_idx'),
+            models.Index(fields=['properties_title'], name='item_title_idx'),
+            models.Index(fields=['geometry'], name='item_geometry_idx'),
+            models.Index(fields=['collection', 'name'], name='item_coll_name_item_name_idx'),
+            models.Index(
+                fields=['properties_datetime', 'properties_start_datetime'],
+                name='item_datetime_start_dttm_idx'
+            ),
+            models.Index(
+                fields=['properties_datetime', 'properties_end_datetime'],
+                name='item_datetime_end_dttm_idx'
+            ),
+            models.Index(
+                fields=[
+                    'properties_datetime', 'properties_start_datetime', 'properties_end_datetime'
+                ],
+                name='item_dttme_start_end_dttm_idx'
+            ),
+        ]
```

Comment:
+1 for the intro and I will also take time to check the EXPLAIN ANALYZE PostgeSQL command.
What I did in this first shot is, to check for all `.filter()` or `Q()` and the like and create indices for the queried fields. I tried to avoid redundancies with autogenerated indices for PK and FK, but I might have missed some and still have redundancies.

I think, an intro by @ltclm / @rebert might be very nice here :-)

Reply:
you can paste the explain analyze output here: https://explain.depesz.com/
it will show you the slow parts of the query and there is lots of background information on the different numbers.
A first step would be to look out for sequential scans, such scans should be avoided whenever possible.

i have no idea about the django / orm limitations concerning index creation in postgres. I would try to analyze the slow queries with postgres / pgadmin, identify the missing indexes and then try to add them to the orm model ( if possible ).